mongo/gridfs/bucket.go

```diff
@@ -255,6 +256,17 @@ func (b *Bucket) Delete(fileID interface{}) error {
 // DeleteContext deletes all chunks and metadata associated with the file with the given file ID and runs the underlying
 // delete operations with the provided context.
 func (b *Bucket) DeleteContext(ctx context.Context, fileID interface{}) error {
+	// If no deadline is set on the passed-in context, Timeout is set on the Client, and context is
```

Comment:
I suggest adding this check to the other new `Context` methods. Consider putting in a helper function.

Reply:
This check is only needed for testing. The way we run tests like [these](https://github.com/mongodb/specifications/blob/master/source/client-side-operations-timeout/tests/gridfs-delete.yml#L78) that set `timeoutMS` for a single GridFS operation is by creating a Timeout context in the [unified test runner](https://github.com/mongodb/mongo-go-driver/blob/master/mongo/integration/unified/operation.go#L93-L105) and running `DeleteContext` with that new context. `DeleteContext` and `DropContext` are unique in the fact that they run _two_ operations in the same method (for the files and chunks collections). We need to use the same Timeout context for both internal operations, so this check is necessary for `DeleteContext` and `DropContext`. The other `Context` methods run only a single operation against the server, so this check is not necessary. Given we only need this check for two methods, I think the use of a helper function will not significantly reduce complexity and may hinder readability, but I do not have a strong opinion.

Reply:
Thank you for the explanation. I agree a helper is not necessary.